samples: nrf9160: modem_shell: add command for heap usage statistics

tokangas opened the pull request:
Added a new command called `heap`, which prints out kernel and system heap usage statistics.

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10211/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10211/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-fw-nrfconnect-nrf-iot_mosh | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10211/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-10211
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

tokangas commented:
I modified the code to use `mallinfo()` function instead of `malloc_stats()` and added printing of maximum system heap size. Also added the `heap` command to the documentation, ping @peknis.

The current output is like this:
```
mosh:~$ heap
kernel heap statistics:
free:             7804
allocated:         272
max. allocated:   1056

system heap statistics:
max. size:       81400
size:              248
free:              160
allocated:          88
```

peknis commented:
> I modified the code to use `mallinfo()` function instead of `malloc_stats()` and added printing of maximum system heap size. Also added the `heap` command to the documentation, ping @peknis.

Still looks good, I thought you would add the command.

trantanen commented on samples/nrf9160/modem_shell/README.rst:
```diff
@@ -714,6 +714,27 @@ Disable UARTs for power measurement purposes.
+Heap usage statistics
+=====================
+
+You can use the the ``heap`` command to print kernel and system heap usage statistics.
```
```suggestion
You can use the ``heap`` command to print kernel and system heap usage statistics.
```

tokangas commented:
@rlubos I guess the kernel heap part would fit also in Zephyr kernel shell, what do you think?

rlubos commented:
> @rlubos I guess the kernel heap part would fit also in Zephyr kernel shell, what do you think?

Yes I think it could be accepted upstream, although it's not really my area so can't tell for sure. Seems pretty useful though.

tokangas commented:
> Yes I think it could be accepted upstream, although it's not really my area so can't tell for sure. Seems pretty useful though.

Ok, I think I'll create a PR for Zephyr. We can merge this anyway and later remove the kernel heap part if we get that into Zephyr.
